assets/js/modules/adsense/components/setup/v2/SetupMain.js

```diff
@@ -0,0 +1,319 @@
+			accountID,
+			clientID,
+		] )
+	);
+
+	const accountStatus = determineAccountStatus( {
+		accounts,
+		clients,
+		alerts,
+		urlChannels,
+		accountsError,
+		alertsError,
+		urlChannelsError,
+		previousAccountID,
+		previousClientID,
+	} );
+	const siteStatus = determineSiteStatus( {
+		urlChannels,
+		siteURL,
+	} );
+
+	useUpdateEffect( () => {
+		if (
+			accountID !== undefined ||
+			clientID !== undefined ||
+			accountStatus !== undefined ||
+			siteStatus !== undefined
+		) {
+			setIsAwaitingBackgroundSubmit( true );
+		}
+	}, [ accountID, clientID, accountStatus, siteStatus ] );
+
+	// Update current account ID setting on-the-fly.
+	useEffect( () => {
+		if (
+			! accountID ||
+			accounts?.length !== 1 ||
+			previousAccountID ||
+			( accounts?.length === 1 &&
+				accounts.findIndex( ( { _id } ) => _id === accountID ) === -1 )
+		) {
+			return;
+		}
+
+		setAccountID( accountID );
+		// Set flag to await background submission.
+		setIsAwaitingBackgroundSubmit( true );
+	}, [ accounts, accountID, previousAccountID, setAccountID ] );
+
+	// Update account status on-the-fly.
+	useEffect( () => {
+		if ( accounts?.length === 0 ) {
+			setAccountStatus( ACCOUNT_STATUS_NONE );
+		} else if ( accounts?.length && ! accountID ) {
+			setAccountStatus( ACCOUNT_STATUS_MULTIPLE );
+		} else {
+			setAccountStatus( ACCOUNT_STATUS_NO_CLIENT );
+		}
+	}, [ setAccountStatus, accountID, accounts ] );
+
+	// If a background submission should happen and changes are valid to be
+	// submitted, do that here. This is wrapped in a separate useEffect hook
+	// and relies on isAwaitingBackgroundSubmit since the above useEffect hook
+	// must not depend on canSubmitChanges, since that is also updated when
+	// other than the above four settings are updated.
+	useEffect( () => {
+		if (
+			! isAwaitingBackgroundSubmit ||
+			isSubmittingInBackground ||
+			! canSubmitChanges
+		) {
+			return;
+		}
+
+		// Set flag to false since we are gonna run the background submission
+		// right now.
+		setIsAwaitingBackgroundSubmit( false );
+
+		// Set internal state for submitting in background to avoid sudden
+		// rendering of a progress bar.
+		( async () => {
+			setIsSubmittingInBackground( true );
+			await submitChanges();
+			setIsSubmittingInBackground( false );
+		} )();
+	}, [
+		isAwaitingBackgroundSubmit,
+		isSubmittingInBackground,
+		canSubmitChanges,
+		submitChanges,
+	] );
+
+	// Reset all fetched data when user re-focuses tab.
+	useEffect( () => {
+		let timeout;
+		let idleSeconds = 0;
+		// Count seconds once user focuses elsewhere.
+		const countIdleTime = () => {
+			timeout = global.setInterval( () => {
+				idleSeconds++;
+			}, 1000 );
+		};
+		// Reset when user re-focuses after 15 seconds or more.
+		const reset = () => {
+			global.clearTimeout( timeout );
+			// Do not reset if user has been away for less than 15 seconds.
+			if ( idleSeconds < 15 ) {
+				idleSeconds = 0;
+				return;
+			}
+			idleSeconds = 0;
```

Comment:
Maybe instead of using counters, it would be easier to use a flag. Something like this:

```js
		let needReset = false;
		// Count seconds once user focuses elsewhere.
		const countIdleTime = () => {
			timeout = global.setTimeout( () => {
				needReset = true;
			}, 15000 ); // 15 seconds
		};
		// Reset when user re-focuses after 15 seconds or more.
		const reset = () => {
			global.clearTimeout( timeout );
			// Do not reset if user has been away for less than 15 seconds.
			if ( ! needReset ) {
				return;
			}
			needReset = false;
```

Reply:
Nice improvement. I just copied over what was in the main Setup file